storm-server/src/main/java/org/apache/storm/scheduler/utils/ArtifactoryConfigLoader.java

```diff
@@ -0,0 +1,416 @@
+            }
+            _lastReturnedTime = Time.currentTimeSecs();
+            _lastReturnedValue = ret;
+            return _lastReturnedValue;
+        }
+
+        return null;
+    }
+
+
+    private Map getLatestFromCache() {
+        String localFileName = _localCacheDir + File.separator + cacheFilename;
+        return loadFromFile(new File(localFileName));
+    }
+
+    private void saveInArtifactoryCache(String yamlData) {
+        String localFileName = _localCacheDir + File.separator + cacheFilename;
+
+        File cacheFile = new File(localFileName);
+        try (FileOutputStream fos = new FileOutputStream(cacheFile)) {
+            fos.write(yamlData.getBytes());
+            fos.flush();
+        } catch (IOException e) {
+            LOG.error("Received exception when writing file {}.  Attempting delete", localFileName, e);
+            try {
+                cacheFile.delete();
+            } catch (Exception deleteException) {
+                LOG.error("Received exception when deleting file {}.", localFileName, deleteException);
+            }
+        }
+    }
+
+    private void makeArtifactoryCache(String location) {
+        // Just make sure approprite directories exist
+        String localDirName = (String)_conf.get(Config.STORM_LOCAL_DIR);
+        if (localDirName == null) {
+            return;
+        }
+
+        // First make the cache dir
+        localDirName = localDirName + File.separator + "nimbus" + File.separator + LOCAL_ARTIFACT_DIR;
+        File dir = new File(localDirName);
+        if (! dir.exists()) {
+            dir.mkdir();
+        }
+
+        _localCacheDir = localDirName + File.separator + location.replaceAll(File.separator, "_");
+        dir = new File(_localCacheDir);
+        if (! dir.exists()) {
+            dir.mkdir();
+        }
+        _cacheInitialized = true;
+    }
+
+    @Override
+    public void prepare(Map conf) {
+        _conf = conf;
+        String thisTimeout = (String)_conf.get(ARTIFACTORY_TIMEOUT_SECS);
+        if (thisTimeout != null) {
+            _timeoutSeconds = Integer.parseInt(thisTimeout);
+        }
+        String thisPollTime = (String)_conf.get(ARTIFACTORY_POLL_TIME_SECS);
+        if (thisPollTime != null) {
+            _artifactoryPollTimeSecs = Integer.parseInt(thisPollTime);
+        }
+        String thisScheme = (String)_conf.get(ARTIFACTORY_SCHEME);
+        if (thisScheme != null) {
+            _artifactoryScheme = thisScheme;
+        }
+        String thisBase = (String)_conf.get(ARTIFACTORY_BASE_DIRECTORY);
+        if (thisBase != null) {
+            _baseDirectory = thisBase;
+        }
+    }
+
+    @Override
+    public Map load() {
+        // Check for new file every so often 
+        if (_lastReturnedValue != null && ((Time.currentTimeSecs() - _lastReturnedTime) < _artifactoryPollTimeSecs)) {
+            LOG.debug("returning our last map");
```

Comment:
Are you saying to add _lastReturnedTime, Time.currentTimeSecs() and _artifactoryPoolTimeSecs to cache file?  Sorry if I mis-understand your comments